tsdb/db.go

```diff
@@ -760,6 +760,9 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 
 func removeBestEffortTmpDirs(l log.Logger, dir string) error {
 	files, err := ioutil.ReadDir(dir)
+	if files == nil {
+		return nil
+	}
```

Comment:
We don't need this check

Reply:
This is not the right check for that issue. You should instead be doing `if os.IsNotExist(err) { return nil }`. Else with `files==nil`, it will ignore any error here.